src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -317,8 +321,9 @@ private void OnFileChanged(FileSystemEventArgs e)
                 }
 
                 TraceFileChangeRestart(changeDescription, e.ChangeType.ToString(), e.FullPath, shutdown);
-                ScheduleRestartAsync(shutdown).ContinueWith(t => _logger.LogError(t.Exception, $"Error restarting host (full shutdown: {shutdown})"),
-                    TaskContinuationOptions.ExecuteSynchronously | TaskContinuationOptions.OnlyOnFaulted);
+                ScheduleRestartAsync("Monitored file change detected.", shutdown)
```

Comment:
Can we list which file it was here also? I think we write it out above, but sometimes there's so much logging happening during this that it's hard to correlate exactly.

Reply:
<img width="998" height="98" alt="image" src="https://github.com/user-attachments/assets/310f46fb-e9f5-4423-beb2-9fe349dc6ddf" />